### PR TITLE
Revert User model primary id change

### PR DESCRIPTION
**Pull request by manhg**
This break existing tokens references, partially 
in SOPS when using new OAuth2

authtoken_token table uses user.id as primary key

[Relates #119429865]

### CODE REVIEW

File: locarise_drf_oauth2_support/oauth2_client/backends.py [hunk 1]
```diff
@@ -36,21 +36,6 @@ def access_token_url(self):
             'https://accounts.locarise.com/oauth2/access_token'
         )
 
-    def get_user_details(self, response):
```
Comment (charles-vdulac) [inline locarise_drf_oauth2_support/oauth2_client/backends.py:39]: This part is required. We need it to fill local user object.

File: locarise_drf_oauth2_support/oauth2_client/views.py [hunk 2]
```diff
@@ -16,6 +17,13 @@
 User = get_user_model()
 
 
+def get_data(request):
```
Comment (charles-vdulac) [inline locarise_drf_oauth2_support/oauth2_client/views.py]: According to the `setup.py` file, the goal of this lib is not to support DRF2 but only DRF3.

File: locarise_drf_oauth2_support/users/serializers.py [hunk 3]
```diff
@@ -17,10 +17,9 @@ class CurrentUserSerializer(serializers.ModelSerializer):
     is_staff = serializers.BooleanField(read_only=True)
     first_name = serializers.CharField(read_only=True)
     last_name = serializers.CharField(read_only=True)
-    locale = serializers.CharField(read_only=True)
```
Comment (charles-vdulac) [inline locarise_drf_oauth2_support/users/serializers.py:20]: Why did you remove this field ?

Comment (manhg) [inline locarise_drf_oauth2_support/users/serializers.py:20]: It's not available on SOPS

File: locarise_drf_oauth2_support/oauth2_client/backends.py (hunk 1 above)
Comment (manhg) [inline locarise_drf_oauth2_support/oauth2_client/backends.py:39]: I saw `from Slack account`, seems unused.

Comment (charles-vdulac) [inline locarise_drf_oauth2_support/oauth2_client/backends.py:39]: Docstring is wrong. Should be `Return user details from Locarise account`

File: locarise_drf_oauth2_support/users/serializers.py (hunk 3 above)
Comment (charles-vdulac) [inline locarise_drf_oauth2_support/users/serializers.py:20]: Add it

Comment (manhg) [inline locarise_drf_oauth2_support/users/serializers.py:20]: I fixed
